core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -41,13 +44,7 @@ private[spark] trait Logging {
   }
 
   // Method to get or create the logger for this object
-  protected def log: Logger = {
-    if (log_ == null) {
-      initializeLogIfNecessary(false)
-      log_ = LoggerFactory.getLogger(logName)
-    }
-    log_
-  }
+  protected def log: Logger = log_
```

Comment:
Fixed via
https://github.com/infynyxx/spark/commit/986e61f96ece6d7f71ff760384cc7553fe3828f4

On Wed, Jun 22, 2016 at 2:17 PM, Sean Owen <notifications@github.com> wrote:

> In core/src/main/scala/org/apache/spark/internal/Logging.scala
> <https://github.com/apache/spark/pull/13842#discussion_r68060626>:
>
> > @@ -41,13 +44,7 @@ private[spark] trait Logging {
> >    }
> >
> >    // Method to get or create the logger for this object
> > -  protected def log: Logger = {
> > -    if (log_ == null) {
> > -      initializeLogIfNecessary(false)
> > -      log_ = LoggerFactory.getLogger(logName)
> > -    }
> > -    log_
> > -  }
> > +  protected def log: Logger = log_
>
> Good point. I think that can be removed. I don't see the point of it, as
> neither it nor its subclass even logs anything directly?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/spark/pull/13842/files/88ffc87c670deb675f443891a4b2e3315467098e#r68060626>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AACdKnkKjiFYLhaEMT3wJslNc4olwNSKks5qOUPtgaJpZM4I7vOl>
> .
>



-- 
--
Cheers,
Praj